numpy/core/setup.py

```diff
@@ -452,17 +452,16 @@ def generate_config_h(ext, build_dir):
                 moredefs.append(('NPY_RELAXED_STRIDES_DEBUG', 1))
 
             # Get long double representation
-            if sys.platform != 'darwin':
-                rep = check_long_double_representation(config_cmd)
-                if rep in ['INTEL_EXTENDED_12_BYTES_LE',
-                           'INTEL_EXTENDED_16_BYTES_LE',
-                           'MOTOROLA_EXTENDED_12_BYTES_BE',
-                           'IEEE_QUAD_LE', 'IEEE_QUAD_BE',
-                           'IEEE_DOUBLE_LE', 'IEEE_DOUBLE_BE',
-                           'DOUBLE_DOUBLE_BE', 'DOUBLE_DOUBLE_LE']:
-                    moredefs.append(('HAVE_LDOUBLE_%s' % rep, 1))
-                else:
-                    raise ValueError("Unrecognized long double format: %s" % rep)
+            rep = check_long_double_representation(config_cmd)
+            if rep in ['INTEL_EXTENDED_12_BYTES_LE',
+                       'INTEL_EXTENDED_16_BYTES_LE',
+                       'MOTOROLA_EXTENDED_12_BYTES_BE',
+                       'IEEE_QUAD_LE', 'IEEE_QUAD_BE',
+                       'IEEE_DOUBLE_LE', 'IEEE_DOUBLE_BE',
+                       'IBM_DOUBLE_DOUBLE_BE', 'DOUBLE_DOUBLE_LE']:
```

Comment:
This check seems pointless, because `long_double_representation` always returns a member from this list or throws an exception.

Reply:
true, removed